src/passes/OptimizeInstructions.cpp

```diff
@@ -3597,6 +3597,57 @@ struct OptimizeInstructions
         }
       }
     }
+    {
+      // copysign(x, +C)   ==>   abs(x)
+      // copysign(x, -C)   ==>   neg(abs(x))
```

Comment:
This one seems slower - two operations instead of one.

Reply:
No, In many architectures nabs is single instruction. See [this](https://github.com/bytecodealliance/wasmtime/issues/4803). While `copysign` required 3-4 instructions.

Reply:
I think sooner or later they will optimize this as well as single instruction. I will create an issue for v8 for this. But even if web engines don't optimize yet it should still be faster than `copysign` or at least with the same speed. While it will definitely have a smaller size since we get rid of the constant (-8 bytes)

Reply:
why would it be the same speed, if they don't optimize it? surely copysign takes as many cycles as abs or neg?

Reply:
That's how copysign lowered on x64 for example:
https://github.com/bytecodealliance/wasmtime/blob/2986f6b0fffd576a2ba88a1fc6cb2db959a1f9e8/cranelift/filetests/filetests/isa/x64/fcopysign.clif#L33

So it's 1 `movabsq` +  2 `movdqa` + 1 `andpd` + 1 `orpd`

Reply:
For example with AVX it could looks like this:
```asm
mov 0x7fffffff, %r10d
vmovd %r10d, %xmm3
vpternlogd $0xe4, %xmm3, %xmm1, %xmm0 
```

Reply:
Interesting, but I don't know enough about this. Does x64 really not have a CPU instruction for copysign? A wasm principle has always been to add wasm instructions only for things that have CPU instructions that correspond to them, in major architectures at least.